### PR TITLE
Add function to calculate ensemble (relative) frequency

### PR DESCRIPTION
**Pull request by rpmanser**
Add function ens_frequency(), which calculates individual binary fields for ensemble members, ensemble frequency, or ensemble relative frequency given a threshold value.

### CODE REVIEW

File: wrf_ens_tools/calc/calc.py [hunk 1]
```diff
@@ -575,6 +575,42 @@ def dist_mask(xind, yind, xpts, ypts, r):
     """
     return (np.sqrt(((xind - xpts)**2) + ((yind - ypts)**2)) <= r)
 
+def ens_frequency(ens_field, thresh, field='counts'):
+    """
+    Calculate ensemble frequency of a given 2-dimensional variable for a specified threshold.
+
+    The number of ensemble members is inferred by the first dimension of `ens_field`.
+
```
Comment (tjwixtrom) [inline wrf_ens_tools/calc/calc.py]: Add an axis argument to specify which axis is the ensemble members. Default can be axis=0

File: wrf_ens_tools/calc/calc.py [hunk 2]
```diff
@@ -575,6 +575,42 @@ def dist_mask(xind, yind, xpts, ypts, r):
...
+    mask = (ens_field >= thresh)
+    members, ypoints, xpoints = np.asarray(mask).nonzero()
+
+    bins = np.zeros_like(ens_field)
+    bins[members, ypoints, xpoints] = 1.
+
+    if field == 'bins':
+        return bins
+    elif field == 'counts':
+        return np.sum(bins, axis=0)
```
Comment (tjwixtrom) [inline wrf_ens_tools/calc/calc.py]: `return np.sum(bins, axis=axis)`

File: wrf_ens_tools/calc/calc.py [hunk 3]
```diff
@@ -575,6 +575,42 @@ def dist_mask(xind, yind, xpts, ypts, r):
...
+    elif field == 'probs':
+        return np.sum(bins, axis=0) / ens_field.shape[0]
```
Comment (tjwixtrom) [inline wrf_ens_tools/calc/calc.py]: `return np.sum(bins, axis=axis) / ens_field.shape[axis]`

**Comment by rpmanser** [comment]
Good call on the axis argument. I added that and updated the docstring.

**Comment by ac0015** [comment]
This is also a good idea! Only suggestion would be to add a neighborhood option for the prob field, but we could do this later if needed. May update the surrogate severe code to use this since it's a lot cleaner.

**Comment by rpmanser** [comment]
> This is also a good idea! Only suggestion would be to add a neighborhood option for the prob field, but we could do this later if needed. May update the surrogate severe code to use this since it's a lot cleaner.

Good point. I plan on writing something to calculate the neighborhood probability field in Python, so we can keep this open until I get that wrapped up, or we can close it now and address that later.

**Comment by ac0015** [comment]
In that case, maybe we can merge this for now and do the neighborhood functions in a separate branch? If you're implementing all that in Python I imagine it'll come with quite a few additions!

**Comment by tjwixtrom** [comment]
Looks good to me, just need @ac0015 to approve and merge.

**Pull request merged**